src/actions/ui.js

```diff
@@ -1,12 +1,18 @@
 // @flow
 const constants = require("../constants");
-const { getSource } = require("../selectors");
+const { getSource, getFileSearchState } = require("../selectors");
 import type { ThunkArgs } from "./types";
 
 function toggleFileSearch(searchOn: boolean) {
-  return {
-    type: constants.TOGGLE_FILE_SEARCH,
-    searchOn
+  return ({ dispatch, getState }: ThunkArgs) => {
+    if (searchOn) {
```

Comment:
Hi @jasonLaster 

I actually tried that....There are 2 problems:
1. `getFileSearchState(getState())` would return `undefined` when 1. first click on new tab button and 2. click on any of the search results.
2. In some cases, such as pressing esc, clicking on the close button for the search bar, etc, we actually don't want to toggle file search. We want to close it.

These are the reasons why I pass `true` to toggle search bar and `false` to close the search bar.